File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/util/SessionUtils.java
package org.javamaster.invocationlab.admin.util;

import org.springframework.web.context.request.RequestContextHolder;
import org.springframework.web.context.request.ServletRequestAttributes;

import javax.servlet.http.HttpSession;
import java.util.Objects;

public class SessionUtils {

    public static <T> void saveToSession(String key, T obj) {
        ServletRequestAttributes requestAttributes = (ServletRequestAttributes) RequestContextHolder.getRequestAttributes();
        HttpSession session = Objects.requireNonNull(requestAttributes).getRequest().getSession();
        session.setAttribute(key, obj);
    }

    @SuppressWarnings("unchecked")
    public static <T> T getFromSession(String key) {
        ServletRequestAttributes requestAttributes = (ServletRequestAttributes) RequestContextHolder.getRequestAttributes();
        HttpSession session = Objects.requireNonNull(requestAttributes).getRequest().getSession();
        return (T) session.getAttribute(key);
    }
}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/util/SessionUtils.kt
package org.javamaster.invocationlab.admin.util

import org.springframework.web.context.request.RequestContextHolder
import org.springframework.web.context.request.ServletRequestAttributes

object SessionUtils {
    @JvmStatic
    fun <T> saveToSession(key: String?, obj: T) {
        val requestAttributes = RequestContextHolder.getRequestAttributes() as ServletRequestAttributes?
        val session = requestAttributes!!.request.session
        session.setAttribute(key, obj)
    }

    @JvmStatic
    fun <T> getFromSession(key: String?): T {
        val requestAttributes = RequestContextHolder.getRequestAttributes() as ServletRequestAttributes?
        val session = requestAttributes!!.request.session
        @Suppress("UNCHECKED_CAST")
        return session.getAttribute(key) as T
    }
}
